controllers/servers/host_definer/watcher/host_definition_watcher.py

```diff
@@ -57,19 +57,27 @@ def _is_host_definition_in_desired_state(self, host_definition):
     def _handle_pending_host_definition(self, host_definition):
         response = DefineHostResponse()
         phase = host_definition.phase
+        action = self._get_action(phase)
         if phase == settings.PENDING_CREATION_PHASE:
             response = self._define_host(host_definition)
         elif self._is_pending_for_deletion_need_to_be_handled(phase, host_definition.node_name):
             response = self._undefine_host(host_definition)
         self._handle_error_message_for_pending_host_definition(
-            host_definition, response.error_message)
+            host_definition, response.error_message, action)
 
-    def _handle_error_message_for_pending_host_definition(self, host_definition, error_message):
+    def _get_action(self, phase):
+        if phase == settings.PENDING_CREATION_PHASE:
+            return settings.DEFINE
+        elif phase == settings.PENDING_DELETION_PHASE:
+            return settings.UNDEFINE
+        return ''
```

Comment:
what is this `action`?

Reply:
the action is what we intended to do, to define the host on the storage, or undefined it from the storage.
We are using this action later in the error event that we will know that this event is related to define or undefined action

Reply:
yes, so what is the action `''`

Reply:
removed this line, it will return define or undefine

Reply:
I don't understand the answer..

Reply:
that I removed the action `''` and now this function only returns define or undefined